pkgs/by-name/ne/nexusmods-app/package.nix

```diff
@@ -32,45 +54,92 @@ buildDotnetModule (finalAttrs: {
   projectFile = "src/NexusMods.App/NexusMods.App.csproj";
   testProjectFile = "NexusMods.App.sln";
 
-  buildInputs = [ avalonia ];
+  buildInputs = [
+    # TODO: bump avalonia to 11.1.3
+    # avalonia
+  ];
 
-  nativeBuildInputs = [ copyDesktopItems ];
+  nativeCheckInputs = [ _7zz ];
+
+  nativeBuildInputs = [
+    copyDesktopItems
+    # TODO: Remove when patch isn't needed
+    dos2unix
+    imagemagick # For resizing SVG icon in postInstall
+  ];
 
   nugetDeps = ./deps.nix;
   mapNuGetDependencies = true;
 
   dotnet-sdk = dotnetCorePackages.sdk_8_0;
   dotnet-runtime = dotnetCorePackages.runtime_8_0;
 
-  postConfigureNuGet = ''
-    dotnet add src/NexusMods.Icons/NexusMods.Icons.csproj package SkiaSharp -v 2.88.7
-  '';
-
-  preConfigure = ''
-    substituteInPlace Directory.Build.props \
-      --replace '</PropertyGroup>' '<ErrorOnDuplicatePublishOutputFiles>false</ErrorOnDuplicatePublishOutputFiles></PropertyGroup>'
+  prePatch = ''
+    # TODO: Remove when patch isn't needed
+    dos2unix src/ArchiveManagement/NexusMods.FileExtractor/build/NexusMods.FileExtractor.targets
   '';
 
   postPatch = ''
-    ln --force --symbolic "${lib.getExe _7zz}" src/ArchiveManagement/NexusMods.FileExtractor/runtimes/linux-x64/native/7zz
+    # TODO: Remove when patch isn't needed
+    unix2dos src/ArchiveManagement/NexusMods.FileExtractor/build/NexusMods.FileExtractor.targets
 
     # for some reason these tests fail (intermittently?) with a zero timestamp
     touch tests/NexusMods.UI.Tests/WorkspaceSystem/*.verified.png
   '';
 
   makeWrapperArgs = [
     "--prefix PATH : ${lib.makeBinPath finalAttrs.runtimeInputs}"
-    # Make associating with nxm links work on Linux
-    "--set APPIMAGE ${placeholder "out"}/bin/NexusMods.App"
   ];
 
-  runtimeInputs = [ desktop-file-utils ];
+  # FIXME: Decide if we should use pname (nexusmods-app) or id (com.nexusmods.app) in filenames
+  # Most desktop entries in nixpkgs seem to use pname, but is that just for historical reasons?
+  # Note: the appstream file lists the "desktop-id" launchable as "com.nexusmods.app.desktop"
+  # Other packages seem to interpolate `pname` in the install paths - should I use `finalAttrs.pname`?
+  postInstall = ''
+    # Desktop entry
+    # As per #308324, use mainProgram from PATH, instead of $out/bin/NexusMods.App
+    install -D -m 444 -t $out/share/applications src/NexusMods.App/com.nexusmods.app.desktop
+    substituteInPlace $out/share/applications/com.nexusmods.app.desktop \
+      --replace-fail '${"$"}{INSTALL_EXEC}' "${finalAttrs.meta.mainProgram}"
```

Comment:
Does anyone have views on this? I'd ideally like to resolve the FIXME before merging.

Reply:
I think I'd personally just leave the filename as it is in $src, but I wouldn't object to any of the options.

Also not important, but for the --replace-fail, I think you might be able to do `'''${INSTALL_EXEC}'` if you prefer.

Reply:
> I think I'd personally just leave the filename as it is in $src, but I wouldn't object to any of the options.

Sounds good. If there's no other immediate objections I'll remove the FIXME comments then.

> Also not important, but for the --replace-fail, I think you might be able to do `'''${INSTALL_EXEC}'` if you prefer.

I just checked this in the repl:
```
nix-repl> ''
          foo '''${"bar"}'
          ''
"foo ''bar'\n"
```

The triple-quotes take priority, so the `$` doesn't get escaped. :sob:

Reply:
Huh, I had no idea about that.  I guess you have to do: `'' ''\'''${foo}' '' `, which is arguably worse.